example/kotlinlib/linting/4-kover/build.mill

```diff
@@ -0,0 +1,60 @@
+package build
+
+import mill._, kotlinlib._
+import kotlinlib.contrib.kover.KoverModule
+
+object `package` extends RootModule with KotlinModule with KoverModule {
+
+  def kotlinVersion = "1.9.24"
+
+  override def koverVersion = "0.8.3"
+
+  object test extends KotlinModuleTests with TestModule.Junit5 with KoverTests {
+    override def forkArgs: T[Seq[String]] = Task {
+      super.forkArgs() ++ Seq("-Dkotest.framework.classpath.scanning.autoscan.disable=true")
+
+    }
+    override def ivyDeps = super.ivyDeps() ++ Agg(
+      ivy"io.kotest:kotest-runner-junit5-jvm:5.9.1"
+    )
+  }
+}
+
+// This is a basic Mill build for a single `KotlinModule`, enhanced with
+// Kover module. The root module extends the `KoverModule` and
+// specifies the version of Kover version to use here: `0.8.3`. This
+// version can be changed if there is a newer one. Now you can call the
+// Kover targets to produce coverage reports.
+// The sub test module extends `KoverTests` to transform the
+// execution of the various testXXX targets to use Kover and produce
+// coverage data.
+// This lets us perform the coverage operations but before that you
+// must first run the test.
+// `./mill test` then `./mill kover.htmlReport` and get your
+// coverage in HTML format.
+//
+// You can download this example project using the *download* link above
+// if you want to try out the commands below yourself. The only requirement is
+// that you have some version of the JVM installed; the `./mill` script takes
+// care of any further dependencies that need to be downloaded.
+
+/** Usage
+
+> ./mill test # Run the tests and produce the coverage data
+...
+... foo.FooTestskotlin - success started
+
+> ./mill resolve kover._ # List what tasks are available to run from kover
+...
+kover.htmlReport
+...
+kover.xmlReport
+...
+
+> ./mill kover.htmlReport
+
+> cat out/kover/htmlReport.dest/kover-report/index.html
```

Comment:
I noticed that the `kover.htmlReport` target gives an incorrect/empty coverage report if `test` is not separately run first. Is that intentional? If that's the same as what `ScoverageModule` and `mill-jacoco` do I guess it's fine to follow suit, just kind of weird

Reply:
I think both expect `test` to be run first indeed. This is what example of `Scoverage` is saying https://github.com/com-lihaoyi/mill/blob/bf81800270ece7adcec07215e1b0bf2735baf107/example/scalalib/linting/2-contrib-scoverage/build.mill#L29-L32 and `mill-jacoco` also runs `test` first: https://github.com/lefou/mill-jacoco/blob/c6d81d370bf780d8b972692fb1f7a7fd335958f6/build.sc#L142-L144.

Do you propose to add some console warning if `test` wasn't run before generating report?

Reply:
How about a console message listing out which test module's coverage data got slurped up, with a big warning if no test module coverage data was detected at all? That would also help mitigate not just the "forgot to run tests at all" case, but also the "ran some tests but not others" failure mode.

Longer term, we should discuss if there's a better way of organizing things, e.g. making the coverage reports have an explicit task dependency on the test coverage data output. But that doesn't need to block this PR which just follows the (imperfect) status quo

Reply:
I added an error log message if binary report is missing: https://github.com/com-lihaoyi/mill/pull/3624/commits/3e00f9d7953ed55ba735e4cea980e05ae25d8957. Not sure how can module name can be reported.